From c53f2106109153eda10c47a6960a5a579d1530b8 Mon Sep 17 00:00:00 2001
From: Gilles <erans@apache.org>
Date: Mon, 1 Dec 2014 14:51:34 +0100
Subject: [PATCH] Deprecation

Deprecating classes whose functionality is based on buggy code in
"BicubicSplinInterpolatingFunction" (MATH-1166).
---
 .../SmoothingPolynomialBicubicSplineInterpolator.java           | 2 ++
 .../interpolation/TricubicSplineInterpolatingFunction.java      | 2 ++
 .../analysis/interpolation/TricubicSplineInterpolator.java      | 2 ++
 3 files changed, 6 insertions(+)

diff --git a/src/main/java/org/apache/commons/math3/analysis/interpolation/SmoothingPolynomialBicubicSplineInterpolator.java b/src/main/java/org/apache/commons/math3/analysis/interpolation/SmoothingPolynomialBicubicSplineInterpolator.java
index 1c76035508..e1639b2f56 100644
--- a/src/main/java/org/apache/commons/math3/analysis/interpolation/SmoothingPolynomialBicubicSplineInterpolator.java
+++ b/src/main/java/org/apache/commons/math3/analysis/interpolation/SmoothingPolynomialBicubicSplineInterpolator.java
@@ -34,7 +34,9 @@
  * polynomial fitting.
  *
  * @since 2.2
+ * @deprecated To be removed in 4.0 (see MATH-1166).
  */
+@Deprecated
 public class SmoothingPolynomialBicubicSplineInterpolator
     extends BicubicSplineInterpolator {
     /** Fitter for x. */
diff --git a/src/main/java/org/apache/commons/math3/analysis/interpolation/TricubicSplineInterpolatingFunction.java b/src/main/java/org/apache/commons/math3/analysis/interpolation/TricubicSplineInterpolatingFunction.java
index 613e70a8fc..4260606572 100644
--- a/src/main/java/org/apache/commons/math3/analysis/interpolation/TricubicSplineInterpolatingFunction.java
+++ b/src/main/java/org/apache/commons/math3/analysis/interpolation/TricubicSplineInterpolatingFunction.java
@@ -34,7 +34,9 @@
  * </quote>
  *
  * @since 2.2
+ * @deprecated To be removed in 4.0 (see MATH-1166).
  */
+@Deprecated
 public class TricubicSplineInterpolatingFunction
     implements TrivariateFunction {
     /**
diff --git a/src/main/java/org/apache/commons/math3/analysis/interpolation/TricubicSplineInterpolator.java b/src/main/java/org/apache/commons/math3/analysis/interpolation/TricubicSplineInterpolator.java
index cda6a33ddb..da19986322 100644
--- a/src/main/java/org/apache/commons/math3/analysis/interpolation/TricubicSplineInterpolator.java
+++ b/src/main/java/org/apache/commons/math3/analysis/interpolation/TricubicSplineInterpolator.java
@@ -26,7 +26,9 @@
  * Generates a tricubic interpolating function.
  *
  * @since 2.2
+ * @deprecated To be removed in 4.0 (see MATH-1166).
  */
+@Deprecated
 public class TricubicSplineInterpolator
     implements TrivariateGridInterpolator {
     /**
